account_chart_update/wizard/wizard_chart_update.py

```diff
@@ -312,779 +187,513 @@ def onchange_company_id(self):
-            tax_codes = tax_code_obj.search(
-                [('name', '=', tax_code_name),
-                 ('company_id', '=', self.company_id.id)])
-        mapping_tax_codes[tax_code_template] = (
-            tax_codes and tax_codes[0] or self.env['account.tax.code'])
-        return mapping_tax_codes[tax_code_template]
+        # Check if errors where detected and wether we should stop.
+        if self.log and not self.continue_on_errors:
+            raise exceptions.Warning(
+                _("One or more errors detected!\n\n%s") % self.log)
+
+        # Store the data and go to the next step.
+        self.state = 'done'
+        return self._reopen()
 
     @api.multi
-    def map_tax_template(self, tax_template, mapping_taxes):
-        """Adds a tax template -> tax id to the mapping."""
-        if not tax_template:
-            return self.env['account.tax']
-        if mapping_taxes.get(tax_template):
-            return mapping_taxes[tax_template]
+    @tools.ormcache("templates")
+    def find_tax_by_templates(self, templates):
+        """Find a tax that matches the template."""
         # search inactive taxes too, to avoid re-creating
         # taxes that have been deactivated before
-        tax_obj = self.env['account.tax'].with_context(active_test=False)
-        criteria = ['|',
-                    ('name', '=', tax_template.name),
-                    ('description', '=', tax_template.name)]
-        if tax_template.description:
-            criteria = ['|', '|'] + criteria
-            criteria += [('description', '=', tax_template.description),
-                         ('name', '=', tax_template.description)]
-        criteria += [('company_id', '=', self.company_id.id)]
-        taxes = tax_obj.search(criteria)
-        mapping_taxes[tax_template] = (
-            taxes and taxes[0] or self.env['account.tax'])
-        return mapping_taxes[tax_template]
+        Tax = self.env['account.tax'].with_context(active_test=False)
+        result = Tax
+        for template in templates:
+            single = Tax
+            criteria = (
+                ("name", "=", template.name),
+                ("description", "=", template.name),
+                ("name", "=", template.description),
+                ("description", "=", template.description),
+            )
+            for domain in criteria:
+                if single:
+                    break
+                if domain[2]:
+                    single = Tax.search(
+                        [domain,
+                         ("company_id", "=", self.company_id.id),
+                         ("type_tax_use", "=", template.type_tax_use)],
+                        limit=1)
+            result |= single
+        return result
+
+    @api.model
+    @tools.ormcache("code")
+    def padded_code(self, code):
+        """Return a right-zero-padded code with the chosen digits."""
+        return "{:0<{}}".format(code, self.code_digits)
```

Comment:
But it's not the case here. Anyway, the same happens with `"Vivo en %s" % "España"` if you don't put the `u`.